From 7dfd10b33284205597c93a69c00a28719ab23fd9 Mon Sep 17 00:00:00 2001
From: Sascha Grunert <sgrunert@suse.com>
Date: Fri, 24 May 2019 10:54:44 +0200
Subject: [PATCH 6/6] Update Makefile to be usable without git

Some build environments will use the CRI-O sources without containing
the complete git history. To avoid annoying fatal git error messages we
now fallback to "unknown" versions if the git repository is not
available.

Signed-off-by: Sascha Grunert <sgrunert@suse.com>
(cherry picked from commit 33af10124c397511723ef6d59dd72edf11353277)
Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
---
 Makefile        | 25 ++++++++++++-------------
 Makefile.inc    | 15 +++++++++++++--
 conmon/Makefile |  4 ++--
 3 files changed, 27 insertions(+), 17 deletions(-)

diff --git a/Makefile b/Makefile
index 295588b..3167232 100644
--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,8 @@
-include Makefile.inc
 
 GO ?= go
 EPOCH_TEST_COMMIT ?= 1cc5a27
 PROJECT := github.com/cri-o/cri-o
-GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
-GIT_BRANCH_CLEAN := $(shell echo $(GIT_BRANCH) | sed -e "s/[^[:alnum:]]/-/g")
-CRIO_IMAGE := crio_dev$(if $(GIT_BRANCH_CLEAN),:$(GIT_BRANCH_CLEAN))
+CRIO_IMAGE = crio_dev$(if $(GIT_BRANCH_CLEAN),:$(GIT_BRANCH_CLEAN))
 CRIO_INSTANCE := crio_dev
 PREFIX ?= ${DESTDIR}/usr/local
 BINDIR ?= ${PREFIX}/bin
@@ -47,11 +44,13 @@ GOPKGBASEDIR := $(shell dirname "$(GOPKGDIR)")
 # Update VPATH so make finds .gopathok
 VPATH := $(VPATH):$(GOPATH)
 SHRINKFLAGS := -s -w
-BASE_LDFLAGS := ${SHRINKFLAGS} -X main.gitCommit=${GIT_COMMIT} -X main.buildInfo=${BUILD_INFO}
-LDFLAGS := -ldflags '${BASE_LDFLAGS}'
+BASE_LDFLAGS = ${SHRINKFLAGS} -X main.gitCommit=${GIT_COMMIT} -X main.buildInfo=${BUILD_INFO}
+LDFLAGS = -ldflags '${BASE_LDFLAGS}'
 
 all: binaries crio.conf docs
 
+include Makefile.inc
+
 default: help
 
 help:
@@ -94,16 +93,16 @@ bin/conmon: conmon/config.h
 bin/pause:
 	$(MAKE) -C pause
 
-test/bin2img/bin2img: .gopathok $(wildcard test/bin2img/*.go)
+test/bin2img/bin2img: git-vars .gopathok $(wildcard test/bin2img/*.go)
 	$(GO) build -i $(LDFLAGS) -tags "$(BUILDTAGS) containers_image_ostree_stub" -o $@ $(PROJECT)/test/bin2img
 
-test/copyimg/copyimg: .gopathok $(wildcard test/copyimg/*.go)
+test/copyimg/copyimg: git-vars .gopathok $(wildcard test/copyimg/*.go)
 	$(GO) build -i $(LDFLAGS) -tags "$(BUILDTAGS) containers_image_ostree_stub" -o $@ $(PROJECT)/test/copyimg
 
-test/checkseccomp/checkseccomp: .gopathok $(wildcard test/checkseccomp/*.go)
+test/checkseccomp/checkseccomp: git-vars .gopathok $(wildcard test/checkseccomp/*.go)
 	$(GO) build -i $(LDFLAGS) -tags "$(BUILDTAGS) containers_image_ostree_stub" -o $@ $(PROJECT)/test/checkseccomp
 
-bin/crio: .gopathok
+bin/crio: git-vars .gopathok
 	$(GO) build -i $(LDFLAGS) -tags "$(BUILDTAGS) containers_image_ostree_stub" -o $@ $(PROJECT)/cmd/crio
 
 crio.conf: bin/crio
@@ -119,7 +118,7 @@ crio.conf: bin/crio
 release-note:
 	@$(GOPATH)/bin/release-tool -n $(release)
 
-conmon/config.h: cmd/crio-config/config.go oci/oci.go
+conmon/config.h: git-vars cmd/crio-config/config.go oci/oci.go
 	$(GO) build -i $(LDFLAGS) -tags "$(BUILDTAGS)" -o bin/crio-config $(PROJECT)/cmd/crio-config
 	( cd conmon && $(CURDIR)/bin/crio-config )
 
@@ -146,14 +145,14 @@ endif
 local-cross:
 	@$(MAKE) --keep-going $(CROSS_BUILD_TARGETS)
 
-bin/crio.cross.%: .gopathok .explicit_phony
+bin/crio.cross.%: git-vars .gopathok .explicit_phony
 	@echo "==> make $@"; \
 	TARGET="$*"; \
 	GOOS="$${TARGET%%.*}" \
 	GOARCH="$${TARGET##*.}" \
 	$(GO) build -i $(LDFLAGS) -tags "containers_image_openpgp btrfs_noversion" -o "$@" $(PROJECT)/cmd/crio
 
-crioimage:
+crioimage: git-vars
 	$(CONTAINER_RUNTIME) build -t ${CRIO_IMAGE} .
 
 dbuild: crioimage
diff --git a/Makefile.inc b/Makefile.inc
index 04149d0..ae04e6a 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -1,2 +1,13 @@
-COMMIT_NO := $(shell git rev-parse HEAD 2> /dev/null || true)
-GIT_COMMIT := $(if $(shell git status --porcelain --untracked-files=no),"${COMMIT_NO}-dirty","${COMMIT_NO}")
+.PHONY: git-vars
+git-vars:
+ifneq ($(wildcard .git),)
+	$(eval COMMIT_NO :=$(shell git rev-parse HEAD 2> /dev/null || true))
+	$(eval GIT_COMMIT := $(if $(shell git status --porcelain --untracked-files=no),"${COMMIT_NO}-dirty","${COMMIT_NO}"))
+	$(eval GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null))
+	$(eval GIT_BRANCH_CLEAN := $(shell echo $(GIT_BRANCH) | sed -e "s/[^[:alnum:]]/-/g"))
+else
+	$(eval COMMIT_NO := unknown)
+	$(eval GIT_COMMIT := unknown)
+	$(eval GIT_BRANCH := unknown)
+	$(eval GIT_BRANCH_CLEAN := unknown)
+endif
diff --git a/conmon/Makefile b/conmon/Makefile
index 0562e38..93cff7f 100644
--- a/conmon/Makefile
+++ b/conmon/Makefile
@@ -1,6 +1,6 @@
-include ../Makefile.inc
+all: git-vars ../bin/conmon
 
-all: ../bin/conmon
+include ../Makefile.inc
 
 src = $(wildcard *.c)
 obj = $(src:.c=.o)
-- 
2.22.0

